bigtop-tests/smoke-tests/hive/HiveJdbcGeneralTest.java

```diff
@@ -0,0 +1,390 @@
+        + " ('Joyce', 'M', 51.3, 50.5, 'BJH', 1),"
+        + " ('Joyce', 'M', 55.5, 53.5, 'BJH', 2),"
+        + " ('Luc', 'M', 66.3, 77.0, 'AJH', 1),"
+        + " ('Luc', 'M', 69.3, 82.9, 'AJH', 2),"
+        + " ('Marie', 'F', 66.5, 112.0, 'BJH', 1),"
+        + " ('Marie', 'F', 69.5, 114.9, 'BJH', 2),"
+        + " ('Medford', 'M', 64.9, 114.0, 'AJH', 1),"
+        + " ('Medford', 'M', NULL, NULL, NULL, NULL),"
+        + " ('Philip', 'M', 69.0, 115.0, 'AJH', 1),"
+        + " ('Philip', 'M', 70.0, 118.0, 'AJH', 2),"
+        + " ('Robert', 'M', 64.8, 128.0, 'BJH', 1),"
+        + " ('Robert', 'M', 68.3, NULL, 'BJH', 2),"
+        + " ('Thomas', 'M', 57.5, 85.0, 'AJH', 1),"
+        + " ('Thomas', 'M', 59.1, 92.3, 'AJH', 2),"
+        + " ('Wakana', 'F', 61.3, 99.0, 'AJH', 1),"
+        + " ('Wakana', 'F', 63.8, 102.9, 'AJH', 2),"
+        + " ('William', 'M', 66.5, 112.0, 'BJH', 1),"
+        + " ('William', 'M', 68.3, 118.2, 'BJH', 2)";
+
+    // failing query from SAS Institute, code generated by PROC MEANS
+    String failingQuery =
+        " select COUNT(*) as ZSQL1, MIN(TXT_1.`school`) as ZSQL2, MIN(TXT_1.`time`) as"
+            + " ZSQL3, COUNT(*) as ZSQL4, COUNT(TXT_1.`height`) as ZSQL5, MIN(TXT_1.`height`)"
+            + " as ZSQL6, MAX(TXT_1.`height`) as ZSQL7, SUM(TXT_1.`height`) as ZSQL8,"
+            + " COALESCE(VAR_SAMP(TXT_1.`height`)*(COUNT(TXT_1.`height`)-1),0) as ZSQL9,"
+            + " COUNT(TXT_1.`weight`) as ZSQL10, MIN(TXT_1.`weight`) as ZSQL11,"
+            + " MAX(TXT_1.`weight`) as ZSQL12, SUM(TXT_1.`weight`) as ZSQL13,"
+            + " COALESCE(VAR_SAMP(TXT_1.`weight`)*(COUNT(TXT_1.`weight`)-1),0) as ZSQL14 from"
+            + " `btestm` TXT_1 where TXT_1.`school` = 'AJH' group by"
+            + " TXT_1.`school`, TXT_1.`time`";
+
+    try (Statement stmt = con.createStatement()) {
+      dropTable(stmt, newTableName + "M");
+      stmt.executeUpdate(
+          "create table btestm (name varchar(8), sex varchar(8), height double, weight double, school varchar(8), `time` double) TBLPROPERTIES('SASFMT:t'='TIME(10.0)')");
+      stmt.executeUpdate(queryValues);
+
+      try (ResultSet res = stmt.executeQuery(failingQuery)) {
+        assertTrue(res.next());
+        assertEquals(null, res.getString(2));
+        ResultSetMetaData rsmd = res.getMetaData();
+        assertEquals(19, rsmd.getPrecision(1));
+        assertEquals(-5, rsmd.getColumnType(1));
+        assertEquals(14, rsmd.getColumnCount());
+        printResults(stmt, "describe formatted btestm");
+        ResultSet res2 = stmt.executeQuery("describe formatted btestm");
+        while (res2.next()) {
+          String section = res2.getString(1);
+          if (section.contains("Detailed"))
+            break;
+        }
+        boolean found = false;
+        while (res2.next()) {
+          String value = res2.getString(2);
+          if (value != null && value.contains("SASFMT"))
+
+          {

```

Comment:
The style here is broken. Reformat using IDE helps.

Reply:
Fixed